src/once_cell.rs

```diff
@@ -427,7 +426,7 @@ impl<T> OnceCell<T> {
 
         // Slow path: initialize the value.
         // The futures provided should never block, so we can use `now_or_never`.
-        now_or_never(self.initialize_or_wait(move || future::ready(closure()), &mut Blocking))?;
+        now_or_never(self.initialize_or_wait(move || async move { closure() }, &mut Blocking))?;
```

Comment:
nit: I'd suggest using `std::future::ready` cause that's more readable IMO.

Reply:
Unfortunately, `std::future::ready` isn't available until Rust version 1.48. I guess we could use it; it doesn't break our general MSRV policy.